src/screens/KeyBasedAssetTransfer/KeyBasedAssetTransferEditAmount.js

```diff
@@ -17,236 +17,114 @@
+      }
+      shouldFooterAvoidKeyboard
     >
-      <ScrollView scrollEnabled={!inSearchMode} contentContainerStyle={{ flex: 1 }}>
-        <SearchBlock
-          searchInputPlaceholder={t('label.searchAsset')}
-          onSearchChange={(query) => setSearchQuery(query)}
-          itemSearchState={searchQuery.length >= 2}
-          navigation={navigation}
-          onSearchFocus={() => setInSearchMode(true)}
-          onSearchBlur={() => setInSearchMode(false)}
+      <Content>
+        <ValueInput
+          value={value}
+          onValueChange={setValue}
+          assetData={assetOption}
+          customAssets={[]}
+          customBalances={keyWalletBalances}
+          onFormValid={setIsValid}
+          getInputRef={(ref) => {
+            inputRef.current = ref;
+          }}
         />
-        <FlatList
-          data={filteredAssetTransfers}
-          scrollEnabled={!inSearchMode}
-          keyExtractor={item => item.assetData.token}
-          renderItem={renderAsset}
-          initialNumToRender={9}
-          contentContainerStyle={{ flexGrow: 1 }}
-          ListEmptyComponent={(
-            <Wrapper
-              flex={1}
-              style={{
-                paddingTop: 90,
-                paddingBottom: 90,
-                alignItems: 'center',
-              }}
-            >
-              <EmptyStateParagraph
-                title={t('transactions.emptyState.assetsList.title')}
-                bodyText={t('transactions.emptyState.assetsList.paragraph')}
-              />
-            </Wrapper>
-          )}
-          refreshControl={
-            <RefreshControl
-              refreshing={isFetchingAvailableBalances}
-              onRefresh={onAvailableBalancesRefresh}
-            />
-          }
-        />
-      </ScrollView>
+      </Content>
     </ContainerWithHeader>
   );
-};
-
-const mapStateToProps = ({
-  keyBasedAssetTransfer: {
-    data: keyBasedAssetsToTransfer,
-    availableBalances,
-    isFetchingAvailableBalances,
-  },
-}: RootReducerState): $Shape<Props> => ({
-  keyBasedAssetsToTransfer,
-  availableBalances,
-  isFetchingAvailableBalances,
-});
-
-const mapDispatchToProps = (dispatch: Dispatch): $Shape<Props> => ({
-  fetchAvailableBalancesToTransfer: () => dispatch(fetchAvailableBalancesToTransferAction()),
-  removeKeyBasedAssetToTransfer: (assetData: AssetData) => dispatch(removeKeyBasedAssetToTransferAction(assetData)),
-  addKeyBasedAssetToTransfer: (assetData: AssetData, amount: number) => dispatch(
-    addKeyBasedAssetToTransferAction(assetData, amount),
-  ),
-});
-
-export default connect(mapStateToProps, mapDispatchToProps)(KeyBasedAssetTransferEditAmount);
+}
+
+export default KeyBasedAssetTransferEditAmount;
+
+const Content = styled.View`
+  flex: 1;
+  padding: ${spacing.largePlus}px 40px ${spacing.large}px;
+`;
+
+const FooterContent = styled.View`
+  width: 100%;
+  padding: ${spacing.small}px ${spacing.large}px ${spacing.large}px;
+`;
```

Comment:
These styles are defined at the bottom of the file, but they're used before they're defined. I know that these days the compiler might not care which way things are ordered (it's not the 90's anymore i guess 😂 ) but it always feels like good practice to define before using. Is there something i'm missing here thought, seriously?

Reply:
This should not be any problem for js engine, as variables (const, lets, etc) are hoisted in JS. So by moving them to the bottom, we can get a clear, easy to read component structure: 
1. imports
2. props type def 
3. component
4. helper functions
5. styles objects & styled components

Reply:
Ahh good old JS hoisting. I was having flashbacks to linter engine rules.